evm/db/shard.py

```diff
@@ -35,75 +36,102 @@ def __init__(self, db):
         self.db = db
 
     #
-    # Collation and header API
+    # Collation Getters by Hash
     #
-    def get_header(self, shard_id: int, period: int) -> CollationHeader:
-        key = make_collation_header_lookup_key(shard_id, period)
+    def get_header_by_hash(self, collation_hash: bytes) -> CollationHeader:
```

Comment:
link to #554 since I think that would introduce the new `Hash32` type for this (depending on what gets merged first.)